src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -0,0 +1,451 @@
+        return sumDocFreq;
+    }
+
+    /**
+     * @return the sum of the term frequencies of all terms in this field across all documents, or -1 if this measurement
+     * isn't available. Term frequency is the total number of occurrences of a term in a particular document and field.
+     *
+     * Note that, documents marked as deleted that haven't yet been merged way aren't taken into account.
+     */
+    public long getSumTotalTermFreq() {
+        return sumTotalTermFreq;
+    }
+
+    /**
+     * @return the lowest value in the field represented as a string.
+     *
+     * Note that, documents marked as deleted that haven't yet been merged way aren't taken into account.
+     */
+    public abstract String getMinValue();
+
+    /**
+     * @return the highest value in the field represented as a string.
+     *
+     * Note that, documents marked as deleted that haven't yet been merged way aren't taken into account.
+     */
+    public abstract String getMaxValue();
+
+    /**
+     * Merges the provided stats into this stats instance.
+     */
+    public void append(FieldStats stats) {
+        this.maxDoc += stats.maxDoc;
+        if (this.docCount == -1 || stats.docCount == -1) {
+            this.docCount = -1;
+        } else {
+            this.docCount += stats.docCount;
+        }
+        if (this.sumDocFreq == -1 || stats.sumDocFreq == -1) {
+            this.sumDocFreq = -1;
+        } else {
+            this.sumDocFreq += stats.sumDocFreq;
+        }
+        if (this.sumTotalTermFreq == -1 || stats.sumTotalTermFreq == -1) {
+            this.sumTotalTermFreq = -1;
+        } else {
+            this.sumTotalTermFreq += stats.sumTotalTermFreq;
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(Fields.MAX_DOC, maxDoc);
+        builder.field(Fields.DOC_COUNT, docCount);
+        builder.field(Fields.DENSITY, getDensity());
+        builder.field(Fields.SUM_DOC_FREQ, sumDocFreq);
+        builder.field(Fields.SUM_TOTAL_TERM_FREQ, sumTotalTermFreq);
+        toInnerXContent(builder);
+        builder.endObject();
+        return builder;
+    }
+
+    protected void toInnerXContent(XContentBuilder builder) throws IOException {
+        builder.field(Fields.MIN_VALUE, minValue);
+        builder.field(Fields.MAX_VALUE, maxValue);
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        maxDoc = in.readVLong();
+        docCount = in.readLong();
+        sumDocFreq = in.readLong();
+        sumTotalTermFreq = in.readLong();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeByte(type);
+        out.writeVLong(maxDoc);
+        out.writeLong(docCount);
+        out.writeLong(sumDocFreq);
+        out.writeLong(sumTotalTermFreq);
+    }
+
+    public static class Lng extends FieldStats<Long> {
```

Comment:
I like to rename it to `Long`, but then it clashes with `java.lang.Long`. Since these concrete classes are kind of private to ES the name shouldn't be an problem? I can also change the generic type to `java.lang.Long`.